charms/admission-webhook/src/charm.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+import os
+from base64 import b64encode
+from pathlib import Path
+from subprocess import run
+
+import yaml
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, MaintenanceStatus
+
+from charmhelpers.core import hookenv
+from oci_image import OCIImageResource, OCIImageResourceError
+
+
+class AdmissionWebhookCharm(CharmBase):
+    def __init__(self, framework, key):
```

Comment:
The "key" here is deprecated, is enough to receive and pass just the "framework".